### PR TITLE
refactor(deps): switch to cosmossdk.io/math from fork math

### PR DESCRIPTION
**Pull request by p0mvn**
Progress towards: https://github.com/osmosis-labs/osmosis/issues/6099

## What is the purpose of the change

This PR proposes creating an alias for `sdkmath` types in `osmomath`.

Primarily, its goal is to remove the need for the verbose and unreadable prefix "Legacy" from `Dec`.

To avoid having to directly import both `sdkmath` and `osmomath` in files where both `Dec` and `Int`/`Uint` types are used we also create aliases for integer types. The aliases are created in the new file `sdk_math_alias.go`

To avoid name clashes with `BigDec` and `BigDec`, their constrictor APIs are now broken to contain "Big" in the suffix. For example, `NewDec` -> `NewBigDec`.

### Important Notes

 * Note: with the update, the Dec and Int do not get initialized to zero values
 by default in proto marhaling/unmarshaling. Instead, they get set to nil values.
 * maxDecBitLen has changed by one bit so overflow panic can be triggerred sooner.
 * Had to update `precompute.go` in CL due to `maxBitLen` changes
 * For converting `Int` to `Dec`, "LegacyDec" is still present since it is impossible to define a method on `Int` outside of `cosmossdk.io/math` package
 * I suggest we do not backport this due to complexity of resolving all these issues and the changes to `maxDecBitLen` + proto marshaling

## Testing and Verifying

This change is already covered by existing tests

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [x] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [x] N/A

### CODE REVIEW

**Review by mattverse — COMMENTED** [review]
Just curious, what's the reason for decision behind making an alias package osmomath and then importing from osmomath directly instead of using sdkmath package?

**Comment by p0mvn** [comment]
> Just curious, what's the reason for decision behind making an alias package osmomath and then importing from osmomath directly instead of using sdkmath package?

This is to reduce verbosity from the `LegacyDec` API stemming from upstream

File: osmomath/decimal.go [hunk 1]
```diff
@@ -75,64 +75,64 @@ func precisionInt() *big.Int {
 	return new(big.Int).Set(precisionReuse)
 }
 
-func ZeroDec() BigDec     { return BigDec{new(big.Int).Set(zeroInt)} }
-func OneDec() BigDec      { return BigDec{precisionInt()} }
-func SmallestDec() BigDec { return BigDec{new(big.Int).Set(oneInt)} }
+func ZeroBigDec() BigDec     { return BigDec{new(big.Int).Set(zeroInt)} }
+func OneBigDec() BigDec      { return BigDec{precisionInt()} }
+func SmallestBigDec() BigDec { return BigDec{new(big.Int).Set(oneInt)} }
```
Comment (mattverse) [inline osmomath/decimal.go:80]: I like these changes, feels like this is what it should have always been

**Review by mattverse — APPROVED** [review]
LGTM, nice work, must have been pain going through all this

**Review by PaddyMc — APPROVED** [review]
Awesome work with this PR, reviewed an it works as expected :+1: 

I have one comment based on the use of `sdk.ZeroDec(` and other uses of the sdk types  package for math functions. There's an [alias created there](https://github.com/osmosis-labs/cosmos-sdk/blob/osmosis-main/types/math.go#L53) that is very similar to the alias created in the `osmomath` package.

When we run `grep -r "sdk.ZeroDec("` we find we refer to the alias in our sdk fork, could we just use `osmomath.ZeroDec(` here instead? And remove the need to import the `sdk "github.com/cosmos/cosmos-sdk/types"` package for math functions?

Approved here with the understanding that this suggestion should probably be another PR if we want to go that route. :zap:

**Comment by p0mvn** [comment]
> grep -r "sdk.ZeroDec("

Seems that some updates got affected by merge conflicts, thanks for catching. Will reswitch everything prior to merge

**Pull request merged**